### ISSUE TITLE
Approach to 'Leave one out split' in case of Sequential Recommenders

### ISSUE THREAD

**Issue by murali-munna** [RUCAIBox/RecBole#1060]
Hi, 
Thanks for the awesome repo. I have two questions/clarifications:

1. In case of sequential recommenders and using `valid_and_test` LS strategy, I understand that the validation results are calculated using the model trained on train data. But will the test results be calculated using data trained on train+validation data? Otherwise, let's say there are `n` points for a user, the `n-1th` point is left-out during testing the `nth` point, which is slightly inaccurate right?

2. What exactly is the difference between `valid_only` and `test_only` strategies? Didn't quite understand from the source code.

Thanks in advance.

**Comment by chenyuwuxin** [comment]
@murali-munna Hello, thanks for your attention to RecBole!

1. For the first question, the test results be calculated don't use data trained on train+validation data. We performed sequence augmentation on the interaction data. 
E.g. Suppose u1 interacts with i1,i2,i3,i4,i5 in time and MAX_ITEM_LIST_LENGTH is 3. After the data augmentation.The sequential dataset becomes {(u1,<i1,0,0>,i2),(u1,<i1,i2,0>,i3),(u1,<i1,i2,i3>,i4),(u1,<i2,i3,i4>,i5)}.In the LS strategy of `valid_and_test` mode,the train dataset, valid dataset and test dataset are {(u1,<i1,0,0>,i2),(u1,<i1,i2,0>,i3)}, {(u1,<i1,i2,i3>,i4)} and {(u1,<i2,i3,i4>,i5)} respectively.

2. For the second question:

- In the LS strategy of `valid_only` mode,the train dataset, valid dataset and test dataset are {(u1,<i1,0,0>,i2),(u1,<i1,i2,0>,i3),(u1,<i1,i2,i3>,i4)} , {(u1,<i2,i3,i4>,i5)} and **None** respectively.Therefore, we will not conduct model test.
- In the LS strategy of `test_only` mode,the train dataset, valid dataset and test dataset are {(u1,<i1,0,0>,i2),(u1,<i1,i2,0>,i3),(u1,<i1,i2,i3>,i4)} , **None** and {(u1,<i2,i3,i4>,i5)} respectively.Therefore, we will not conduct model validation.

I hope my answer can help you.

**Comment by murali-munna** [comment]
@chenyuwuxin Thanks for the detailed explanation. So, for correct results on test set `{(u1,<i2,i3,i4>,i5)}`, we should preferably use `test_only` strategy after fixing the hyperparameters using `valid_and_test` strategy.

**Comment by chenyuwuxin** [comment]
@murali-munna Thank you again for your feedback, we will consider it carefully.
At present, you can set `test_only` mode in your `.yaml` file:
```
eval_args:
  split: {'LS': 'test_only'}
```
You can also refer to [Docs](https://recbole.io/docs/user_guide/config/evaluation_settings.html) to learn more about us.
